classlib/src/main/java/org/teavm/classlib/java/util/TSpliterators.java

```diff
@@ -0,0 +1,120 @@
+/*
+ *  Copyright 2017 Alexey Andreev.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.teavm.classlib.java.util;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.function.Consumer;
+import java.util.function.IntConsumer;
+
+public class TSpliterators {
+
+    private TSpliterators() {
+    }
+
+    public static <T> TSpliterator<T> spliterator(Object[] array, int additionalCharacteristics) {
+        return new TSpliterator<T>() {
+            private int index;
+
+            @Override
+            public boolean tryAdvance(Consumer<? super T> action) {
+                if (index >= array.length) {
+                    return false;
+                }
+
+                action.accept((T) array[index]);
+                index++;
+                return true;
+            }
+
+            @Override
+            public TSpliterator<T> trySplit() {
+                return null;
+            }
+
+            @Override
+            public long estimateSize() {
+                return array.length;
+            }
+
+            @Override
+            public int characteristics() {
+                return additionalCharacteristics | TSpliterator.SIZED;
+            }
+        };
+    }
+
+    public static <T> TSpliterator<T> spliterator(Collection<? extends T> c, int characteristics) {
+        return ((TCollection) c).spliterator();
+    }
+
+    public static TSpliterator.OfInt spliterator(int[] array, int additionalCharacteristics) {
+        return spliterator(array, 0, array.length, additionalCharacteristics);
+    }
+
+    public static TSpliterator.OfInt spliterator(int[] array, int fromIndex, int toIndex,
+                                                 int additionalCharacteristics) {
```

Comment:
FYI: TeaVM code base does not follow this convention with indentation of parameters on single column, use 8 spaces instead.